<a href="https://colab.research.google.com/github/KxxHyoRim/Data-Mining/blob/main/Netflix.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

## Google Drive 사용권한

In [2]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [3]:
#불러올 파일의 경로를 filename 변수에 저장
file = '/content/drive/My Drive/Data Mining/netflix.csv'

#pandas read_csv로 불러오기
df = pd.read_csv(file, encoding='ISO-8859-1')
print(df.shape)
df.head(3)

(5870, 27)


show_id     type  ... Unnamed: 25 Unnamed: 26
0  81145628.0    Movie  ...         NaN         NaN
1  80117401.0    Movie  ...         NaN         NaN
2  70234439.0  TV Show  ...         NaN         NaN

[3 rows x 27 columns]

## Null 값 가진 Column 제거

In [4]:
base_str = 'Unnamed: '

for i in range(12 ,26+1):
  drop_str = base_str + str(i)
  df = df.drop([drop_str], axis=1)

df.head(3)

show_id  ...                                        description
0  81145628.0  ...  Before planning an awesome wedding for his gra...
1  80117401.0  ...  Jandino Asporaat riffs on the challenges of ra...
2  70234439.0  ...  With the help of three human allies, the Autob...

[3 rows x 12 columns]

## Null 값 가진 Row 제거

In [5]:
df['date_added'].isnull()

0       False
1       False
2       False
3       False
4       False
        ...  
5865     True
5866     True
5867     True
5868     True
5869     True
Name: date_added, Length: 5870, dtype: bool

In [6]:
df['date_added']

0       09-Sep-19
1       09-Sep-16
2       08-Sep-18
3       08-Sep-18
4       08-Sep-17
          ...    
5865          NaN
5866          NaN
5867          NaN
5868          NaN
5869          NaN
Name: date_added, Length: 5870, dtype: object

> 2000 row 부터 null


In [7]:
print(df.loc[1999])
print(df.loc[2000]) 

show_id                                               8.00499e+07
type                                                      TV Show
title                                              Five Came Back
director                                        Laurent Bouzereau
cast            Steven Spielberg, Guillermo del Toro, Lawrence...
country                                             United States
date_added                                              31-Mar-17
release_year                                                 2017
rating                                                      TV-MA
duration                                                 1 Season
listed_in                                              Docuseries
description     Five acclaimed contemporary directors tell the...
Name: 1999, dtype: object
show_id         NaN
type            NaN
title           NaN
director        NaN
cast            NaN
country         NaN
date_added      NaN
release_year    NaN
rating          NaN
du

위 내용을 보았을때, idx 1999 까지는 올바르게 값이 저장되어 있지만 <br>
2000부터 5869까지의 row에는 Null값이 들어가있는 것을 확인

In [8]:
'''Null 값의 Row 제거'''

end_idx = df.shape[0]
for i in range(2000, end_idx):
  df = df.drop(i)

In [9]:
df.shape

(2000, 12)

In [10]:
df.tail()   # 마지막 값들에 null 없음

show_id  ...                                        description
1995  80170381.0  ...  A hungry fox and three resourceful, surprising...
1996  80142103.0  ...  En route to a fresh start in Los Angeles, youn...
1997  80179907.0  ...  Performing stand-up for a packed house in Lond...
1998  80152842.0  ...  A young couple fights supernatural foes in an ...
1999  80049928.0  ...  Five acclaimed contemporary directors tell the...

[5 rows x 12 columns]

## 'date_added' Column에서 연도 정보만 추출하기

In [11]:
# date_added 는 csv의 'date_added' 값만 가진 새로운 pandas dataframe
date_added = df['date_added'][:].values

date_added_year = []
idx = 0
for y in date_added:

   #연도 정보 ex. 19 를 가져와서 2000을 더한 뒤 저장
  try:
    date_added_year.append(int(str(y).split('-')[2]) + 2000)
  
  #예외값 처리 -> drop
  except:     
    print(str(idx) + "행에서 Exception 발생 : '" + y + "' 값을 drop합니다")
    df = df.drop(idx)
    date_added = np.delete(date_added, idx)

  idx += 1

print("\nreset index 실행")
df = df.reset_index()  #실행 전 : drop된 row 전체가 index가 유지된 상태로 null로 남아있음

767행에서 Exception 발생 : '2019' 값을 drop합니다
811행에서 Exception 발생 : '2019' 값을 drop합니다

reset index 실행


In [12]:
#연도 정보만 추출이 된 것을 확인할 수 있다.
print(date_added_year)

[2019, 2016, 2018, 2018, 2017, 2017, 2017, 2017, 2017, 2017, 2017, 2017, 2017, 2017, 2017, 2017, 2017, 2017, 2017, 2017, 2017, 2017, 2017, 2017, 2017, 2015, 2018, 2018, 2018, 2018, 2018, 2018, 2018, 2017, 2019, 2019, 2019, 2019, 2019, 2019, 2018, 2018, 2017, 2016, 2019, 2018, 2018, 2018, 2018, 2018, 2018, 2018, 2017, 2017, 2017, 2019, 2019, 2019, 2019, 2018, 2018, 2017, 2019, 2018, 2018, 2018, 2018, 2018, 2018, 2018, 2018, 2017, 2017, 2016, 2018, 2018, 2018, 2018, 2017, 2017, 2017, 2017, 2017, 2017, 2017, 2015, 2019, 2019, 2019, 2018, 2018, 2018, 2018, 2018, 2018, 2018, 2018, 2018, 2017, 2019, 2019, 2019, 2019, 2019, 2019, 2019, 2018, 2017, 2011, 2019, 2019, 2019, 2018, 2017, 2017, 2014, 2019, 2019, 2019, 2019, 2019, 2019, 2018, 2017, 2017, 2019, 2019, 2018, 2017, 2017, 2019, 2019, 2019, 2016, 2016, 2019, 2019, 2018, 2018, 2017, 2017, 2015, 2019, 2019, 2019, 2019, 2019, 2018, 2018, 2018, 2018, 2018, 2018, 2018, 2019, 2019, 2019, 2019, 2019, 2019, 2019, 2019, 2019, 2019, 2019, 2018, 201

In [13]:
print("'date_added' 와 'releas_year'의 shape\n")

print("date_added = " + str(len(date_added_year)))
print( "releas_year = " + str(df['release_year'].shape))

'date_added' 와 'releas_year'의 shape

date_added = 1998
releas_year = (1998,)


In [14]:
type(date_added_year)

list

## DataFrame Merge
두개의 열을 합치기 위해 list 타입의 date_added_year을 datafram으로 변환 

In [15]:
date_added_year = pd.DataFrame(date_added_year, columns=['date_added_year'])

### **상관계수 추출을 위해 result_df 생성**

> result_df = date_added_year와 release_year의 정보만 있음

In [16]:
result_df = pd.concat([date_added_year,df['release_year']],axis=1)
result_df

date_added_year release_year
0                2019         2019
1                2016         2016
2                2018         2013
3                2018         2016
4                2017         2017
...               ...          ...
1993             2018         2015
1994             2017         2017
1995             2017         2016
1996             2017         2016
1997             2017         2017

[1998 rows x 2 columns]

In [17]:
result_df.corr()

date_added_year
date_added_year              1.0

**correlation 파악을 위해 corr() 메소드를 호출했으나** <br>
**release_year의 데이터가 인식되지 않았다.** <br><br>

**Type 출력을 해보니 str 타입이었고, numeric 형식으로 변경 후 정상적으로 corr()을 파악할 수 있었다.**

In [18]:
print(type(result_df['release_year'][0]))
print(type(result_df['date_added_year'][0]))

# Type 변환
result_df['release_year'] = pd.to_numeric(result_df['release_year'])

print()
print(type(result_df['release_year'][0]))
print(type(result_df['date_added_year'][0]))

<class 'str'>
<class 'numpy.int64'>

<class 'numpy.int64'>
<class 'numpy.int64'>


In [19]:
result_df.corr()

date_added_year  release_year
date_added_year         1.000000      0.065335
release_year            0.065335      1.000000